src/router.ts

```diff
@@ -191,25 +240,27 @@ export class Router {
   *
   * @param callbackOrConfig The [[RouterConfiguration]] or a callback that takes a [[RouterConfiguration]].
   */
-  configure(callbackOrConfig: RouterConfiguration|((config: RouterConfiguration) => RouterConfiguration)): Promise<void> {
+  configure(callbackOrConfig: RouterConfiguration | ((config: RouterConfiguration) => RouterConfiguration)): Promise<void> {
     this.isConfigured = true;
 
-    let result = callbackOrConfig;
-    let config;
+    let result: RouterConfiguration = callbackOrConfig as RouterConfiguration;
```

Comment:
It might be a config, or a function that returns void, config, promise<void>, promise<config>. Might be easiest to skip typing.

Reply:
@bigopon